services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/S3CrtAsyncHttpClient.java

```diff
@@ -0,0 +1,255 @@
+import software.amazon.awssdk.utils.AttributeMap;
+import software.amazon.awssdk.utils.http.SdkHttpUtils;
+
+/**
+ * An implementation of {@link SdkAsyncHttpClient} that uses an CRT S3 HTTP client {@link S3Client} to communicate with S3.
+ * Note that it does not work with other services
+ */
+@SdkInternalApi
+public final class S3CrtAsyncHttpClient implements SdkAsyncHttpClient {
+
+    private final S3Client crtS3Client;
+    private final S3NativeClientConfiguration s3NativeClientConfiguration;
+
+    private S3CrtAsyncHttpClient(Builder builder) {
+        s3NativeClientConfiguration =
+            S3NativeClientConfiguration.builder()
+                                       .targetThroughputInGbps(builder.targetThroughputInGbps)
+                                       .partSizeInBytes(builder.minimalPartSizeInBytes)
+                                       .maxConcurrency(builder.maxConcurrency)
+                                       .signingRegion(builder.region == null ? null : builder.region.id())
+                                       .credentialsProvider(builder.credentialsProvider)
+                                       .build();
+
+        S3ClientOptions s3ClientOptions =
+            new S3ClientOptions().withRegion(s3NativeClientConfiguration.signingRegion())
+                                 .withCredentialsProvider(s3NativeClientConfiguration.credentialsProvider())
+                                 .withClientBootstrap(s3NativeClientConfiguration.clientBootstrap())
+                                 .withPartSize(s3NativeClientConfiguration.partSizeBytes())
+                                 .withThroughputTargetGbps(s3NativeClientConfiguration.targetThroughputInGbps());
+        this.crtS3Client = new S3Client(s3ClientOptions);
+    }
+
+    @SdkTestInternalApi
+    S3CrtAsyncHttpClient(S3Client crtS3Client, S3NativeClientConfiguration nativeClientConfiguration) {
+        this.crtS3Client = crtS3Client;
+        this.s3NativeClientConfiguration = nativeClientConfiguration;
+    }
+
+    @Override
+    public CompletableFuture<Void> execute(AsyncExecuteRequest request) {
+        CompletableFuture<Void> executeFuture = new CompletableFuture<>();
+        HttpRequest httpRequest = toCrtRequest(request);
+        S3CrtResponseHandlerAdapter responseHandler = new S3CrtResponseHandlerAdapter(executeFuture, request.responseHandler());
+
+        S3MetaRequestOptions.MetaRequestType requestType = requestType(request);
+
+        S3MetaRequestOptions requestOptions = new S3MetaRequestOptions()
+            .withHttpRequest(httpRequest)
+            .withMetaRequestType(requestType)
+            .withResponseHandler(responseHandler);
+
+        try (S3MetaRequest s3MetaRequest = crtS3Client.makeMetaRequest(requestOptions)) {
+            closeResourcesWhenComplete(executeFuture, s3MetaRequest);
+        }
+
+        return executeFuture;
+    }
+
+    @Override
+    public String clientName() {
+        return "s3crt";
```

Comment:
Does this follow our normal naming convention? I see other names like `NettyNio`, `AwsCommonRuntime`, `Apache`, etc.